File: HomeView.swift
import SwiftUI

struct HomeView: View {
    @State private var isLoggedIn = false
    
    var body: some View {
        NavigationView {
            VStack {
                VStack {
                    Image(systemName: "scalemass")
                        .resizable()
                        .scaledToFit()
                        .frame(width: 80, height: 80)
                        .padding(.bottom, 30)

                    Text("Legal Hub")
                        .font(.system(size: 34, weight: .bold, design: .default))
                        .padding(.bottom, 20)
                }
                .padding(.top, 40)

                VStack(alignment: .leading) {
                    Text("Welcome to Legal Hub")
                        .font(.title2)
                        .fontWeight(.semibold)
                        .padding(.bottom, 10)

                    Text("Connecting you to the best legal professionals.")
                        .font(.subheadline)
                        .foregroundColor(.gray)
                        .lineLimit(1)
                        .padding(.bottom, 30)
                }
                .padding(.horizontal, 50)

                VStack(spacing: 20) {
                    if isLoggedIn {
                        NavigationLink(destination: MainPageView()) {
                            StandardButton(title: "View Homepage", color: .blue)
                        }
                    } else {
                        NavigationLink(destination: LoginView()) {
                            StandardButton(title: "Login", color: .blue)
                        }

                        NavigationLink(destination: RegisterView(), label: {
                            StandardButton(title: "Register", color: .green)
                        })


                        Divider()
                            .padding()

                        Button(action: {
                            // Handle login with Apple action
                            isLoggedIn = true
                        }) {
                            HStack {
                                Image(systemName: "applelogo")
                                    .resizable()
                                    .frame(width: 20, height: 20)
                                    .foregroundColor(.white)

                                Spacer()

                                Text("Continue with Apple")
                                    .font(.headline)
                                    .foregroundColor(.white)
                                
                                Spacer()
                            }
                            .padding()
                            .background(Color.black)
                            .cornerRadius(15)
                        }
                        .frame(maxWidth: .infinity)

                        Button(action: {
                            // Handle login with Google action
                            isLoggedIn = true
                        }) {
                            HStack {
                                Image(systemName: "g.circle.fill")
                                    .resizable()
                                    .frame(width: 20, height: 20)
                                    .foregroundColor(.white)
                                
                                Spacer()

                                Text("Continue with Google")
                                    .font(.headline)
                                    .foregroundColor(.white)

                                Spacer()
                            }
                            .padding()
                            .background(Color.red)
                            .cornerRadius(15)
                        }
                        .frame(maxWidth: .infinity)
                    }
                }
                .padding(.horizontal, 16)

                Spacer()
            }

            .navigationBarItems(leading:
                Button(action: {
                    // Handle language change action
                }) {
                    Image(systemName: "globe")
                        .resizable()
                        .frame(width: 20, height: 20)
                }
            )
        }
    }
}

struct StandardButton: View {
    let title: String
    let color: Color

    var body: some View {
        Text(title)
            .font(.headline)
            .fontWeight(.medium)
            .foregroundColor(.white)
            .padding()
            .frame(maxWidth: .infinity)
            .background(color)
            .cornerRadius(15)
    }
}

File: estsheerniFrontend/src/screens/HomeView.js
import React, { useState } from 'react';
import { Text, Image, View, TouchableOpacity, StyleSheet } from 'react-native';

const CustomButton = ({title, color, onPress}) => (
    <TouchableOpacity onPress={onPress} style={[styles.button, {backgroundColor: color}]}>
        <Text style={styles.buttonText}>{title}</Text>
    </TouchableOpacity>
);

const HomeView = () => {
    const [isLoggedIn, setIsLoggedIn] = useState(false);

    return (
        <View style={styles.container}>
            <Image source={require('./path_to_your_image.png')} style={styles.image} />
            <Text style={styles.header}>Legal Hub</Text>

            <View style={styles.textContainer}>
                <Text style={styles.title}>Welcome to Legal Hub</Text>
                <Text style={styles.subtitle}>Connecting you to the best legal professionals.</Text>
            </View>

            {isLoggedIn ? (
                <CustomButton title="View Homepage" color="#406E9F" onPress={() => {}} />
            ) : (
                <View>
                    <CustomButton title="Login" color="#406E9F" onPress={() => {}} />
                    <CustomButton title="Register" color="#2E7D32" onPress={() => {}} />

                    <View style={styles.divider} />

                    <CustomButton title="Continue with Apple" color="#000" onPress={() => setIsLoggedIn(true)} />
                    <CustomButton title="Continue with Google" color="#4285F4" onPress={() => setIsLoggedIn(true)} />
                </View>
            )}
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        padding: 16,
        alignItems: 'center',
        backgroundColor: '#F5F5F5',
    },
    image: {
        width: 80,
        height: 80,
        marginBottom: 30,
    },
    header: {
        fontSize: 34,
        fontWeight: 'bold',
        marginBottom: 20,
        color: '#406E9F',
    },
    textContainer: {
        width: '80%',
        marginBottom: 30
    },
    title: {
        fontSize: 20,
        fontWeight: 'bold',
        marginBottom: 10,
        color: '#2E7D32',
    },
    subtitle: {
        fontSize: 16,
        color: 'gray',
        marginBottom: 10
    },
    divider: {
        borderBottomColor: 'black',
        borderBottomWidth: 1,
        marginBottom: 10,
        width: '100%',
    },
    button: {
        borderRadius: 10,
        padding: 15,
        marginBottom: 10,
        width: '100%',
        alignItems: 'center',
    },
    buttonText: {
        fontSize: 18,
        color: '#fff',
    },
});

export default HomeView;
